### PR TITLE
Update continuous-delivery-faqs.md

### PR DESCRIPTION
**Pull request by jmoon-harness**
8 FAQs

Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 1]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
 
 #### What is Harness' pipeline execution history retention policy?
 Harness will maintain pipeline execution data for 6 months. You can refer to our [Data Retention](https://developer.harness.io/docs/platform/references/data-retention/#:~:text=Pipeline%20execution%20data%20is%20stored,plan%20you%20are%20subscribed%20to) documentation for more information.
+
+#### How to dynamically create steps (in parallel) based on condition?
+Use case looping strategy, as you can use expressions as well for the strategy condition and those expressions can be generated dynamically.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6621]:
```suggestion
Use case looping strategy as you can use expressions for the strategy condition. These expressions can be generated dynamically.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 2]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+
+#### Move template into git folders
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6623]:
```suggestion
#### How to move template into Git folders?
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 3]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
 Harness will maintain pipeline execution data for 6 months. You can refer to our [Data Retention](https://developer.harness.io/docs/platform/references/data-retention/#:~:text=Pipeline%20execution%20data%20is%20stored,plan%20you%20are%20subscribed%20to) documentation for more information.
+
+#### How to dynamically create steps (in parallel) based on condition?
+Use case looping strategy, as you can use expressions as well for the strategy condition and those expressions can be generated dynamically.
+
+#### Move template into git folders 
+Changing the path from the Harness UI for any entity doesn’t commit the entity on the new path. It will only change the link to the filepath for that entity. So, if you change filepath from A to B, then when loading the entity, it will load the filepath B instead of A whenever you use it. 
+
+#### Is there any validation for moving templates between Git folders?
+There is no validation on any file at B exists or not. It will just try to load and present the expected experience.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6627]:
```suggestion
There is no validation on any file at any folder in Git. It loads and presents the expected experience.
```

Comment (roshnisarangadharan) [reply kb/continuous-delivery/continuous-delivery-faqs.md:6627]: It refers to? And what is the expected experience? Can we be more clear?

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 4]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+
+#### What to do when seeing 500 error when pipeline already exists.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6629]:
```suggestion
#### What to do when you see the 500 error for a pipeline already exists?
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 5]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+The 500 error message can occur when a pipeline URL is pointing to a pipeline identifier with the wrong case. Check the provision identifier to see if the case is in the correct case.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6630]:
```suggestion
The 500 error can occur when a pipeline URL is pointing to a pipeline identifier with the wrong case. Check the provision identifier to see if the case is in the correct.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 6]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+
+#### Can you extract build number from pipeline?
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6632]:
```suggestion
#### Can you extract build number from pipelines?
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 7]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+There is no reference in the pipeline to the Build ID at this time
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6633]:
```suggestion
There is no reference to the Build ID in a pipeline currently.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 8]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+
+#### Why am I getting Error Message: “Invalid request: Trying to run more than 256 concurrent stages/steps. Please upgrade your plan to Enterprise (Paid) or reduce concurrency”?
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6635]:
```suggestion
#### Why am I getting the error message, _Invalid request: Trying to run more than 256 concurrent stages/steps. Please upgrade your plan to Enterprise (Paid) or reduce concurrency_?
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 9]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+This error message occurs when your pipeline has more than the hard limit of 256 steps. This can be caused by a script that is creating these steps.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6636]:
```suggestion
This error message appears when your pipeline has more than 256 steps. This error is caused by a script that creates these steps. 
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 10]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+#### What to do when seeing 500 error when pipeline already exists.
+The 500 error message can occur when a pipeline URL is pointing to a pipeline identifier with the wrong case. Check the provision identifier to see if the case is in the correct case.
+
+#### Can you extract build number from pipeline?
+There is no reference in the pipeline to the Build ID at this time
+
+#### Why am I getting Error Message: “Invalid request: Trying to run more than 256 concurrent stages/steps. Please upgrade your plan to Enterprise (Paid) or reduce concurrency”?
+This error message occurs when your pipeline has more than the hard limit of 256 steps. This can be caused by a script that is creating these steps. 
+
+#### How do I pass output variables from one pipeline stage into the other pipeline?
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6638]:
```suggestion
#### How do I pass output variables from one pipeline stage into other pipeline?
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 11]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+To pass output variables from one pipeline stage to another, you can use pipeline chaining. In the parent pipeline, define the output variable in the output section of the first stage. Then, in the second stage, use the expression <+pipeline.[pipeline_stage_identifier].[output_variable_defined_under_output_section]> to reference the output variable from the first stage. When you run the parent pipeline, the output variable from the first stage will be passed to the second stage as an input variable.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6639]:
```suggestion
To pass output variables from one pipeline stage to another, you can use pipeline chaining. In the parent pipeline, define the output variable in the output section of the first stage. Then, in the second stage, use the expression, `<+pipeline.[pipeline_stage_identifier].[output_variable_defined_under_output_section]>` to reference the output variable from the first stage. When you run the parent pipeline, the output variable from the first stage will be passed to the second stage as an input variable.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 12]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+
+#### Terraform Stage fails saying “failed to find plan”.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6641]:
```suggestion
#### Terraform stage fails saying “failed to find plan”.
```

Comment (roshnisarangadharan) [reply kb/continuous-delivery/continuous-delivery-faqs.md:6641]: Does "failed to find plan" appear in the UI? If yes, is F upper case? I mean "Failed to find plan"? Let's use what appears in the UI.

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 13]
```diff
@@ -6616,3 +6616,27 @@ This error occurs because there's a misconfiguration with the pipeline. Harness
...
+Within Harness, the Terraform stage requires both the Plan and Apply steps in the same stage to properly trigger.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6642]:
```suggestion
Within Harness, the Terraform stage requires both the Plan and Apply steps in the same stage to trigger properly.
```

**Review by roshnisarangadharan — COMMENTED** [review]
Added some suggestions. Could you please resolve merge conflicts, and fix the broken link error in the PR too? Let me know if you need help with doing any of these tasks.

**Comment by aimurphy** [comment]
merging due to age to avoid further build up of MCs.

**Pull request merged**